recipes-ni/sysconfig-settings/sysconfig-settings_1.0.bb

```diff
@@ -122,13 +120,20 @@ do_install () {
 	chown ${LVRT_USER}:${LVRT_GROUP} ${D}${systemsettingsdir}/
 	install -m 0644 ${S}/systemsettings/* ${D}${systemsettingsdir}/
 
-	install -d -m 0755 ${D}${sysconfdir}/natinst/
-	install -m 0755 ${S}/niselectsystemsettings ${D}${sysconfdir}/natinst/
-
 	install -d ${D}${sysconfdir}/init.d/
 	install -m 0755 ${WORKDIR}/nisetembeddeduixml ${D}${sysconfdir}/init.d
 }
 
 pkg_postinst_ontarget_${PN} () {
-	/etc/natinst/niselectsystemsettings postinst
+	TARGET_CLASS=$(/sbin/fw_printenv -n TargetClass 2>&1)
+
+	# Ethernet RIO targets should not have RT startup settings
+	if [ "$TARGET_CLASS" = "Ethernet RIO" ]; then
+		echo "" > ${systemsettingsdir}/rt_target.ini
+	fi
+
+	# cDAQ targets should not have FPGA startup settings
+	if [ "$TARGET_CLASS" = "cDAQ" -o "$TARGET_CLASS" = "CVS" ]; then
```

Comment:
Are all these `TARGET_CLASS`es actually supported in hardknott? If not, can we afford to drop some of this logic?

Reply:
Ethernet RIO is probably droppable here. Trying to confirm on CVS. cDAQ should stay.

I'd really rather these checks be against available hardware instead of target class, but I'm not sure how to detect some of these generically without, for example, enumerating the PCI bus looking for a LabVIEW-targetable FPGA on the bus with one of a growing list of PIDs. Yuck.